fvm/evm/evm_test.go

```diff
@@ -1001,38 +956,24 @@ func TestEVMAddressDeposit(t *testing.T) {
 			bal := getEVMAccountBalance(t, ctx, vm, snapshot, addr)
 			require.Equal(t, expectedBalance, bal)
 
-			// block executed event, make sure TotalGasUsed is non-zero
-			blockEvent := output.Events[3]
-
-			assert.Equal(
-				t,
-				common.NewAddressLocation(
-					nil,
-					common.Address(sc.EVMContract.Address),
-					string(types.EventTypeBlockExecuted),
-				).ID(),
-				string(blockEvent.Type),
-			)
-
-			ev, err := ccf.Decode(nil, blockEvent.Payload)
-			require.NoError(t, err)
-			cadenceEvent, ok := ev.(cadence.Event)
-			require.True(t, ok)
-
-			blockEventPayload, err := types.DecodeBlockEventPayload(cadenceEvent)
-			require.NoError(t, err)
-			require.NotEmpty(t, blockEventPayload.Hash)
-			require.Equal(t, uint64(21000), blockEventPayload.TotalGasUsed)
-
 			// deposit event
-			depositEvent := output.Events[4]
+			depositEvent := output.Events[3]
 			depEv, err := types.FlowEventToCadenceEvent(depositEvent)
 			require.NoError(t, err)
 
 			depEvPayload, err := types.DecodeFLOWTokensDepositedEventPayload(depEv)
 			require.NoError(t, err)
 
 			require.Equal(t, types.OneFlow, depEvPayload.BalanceAfterInAttoFlow.Value)
+
+			// commit block
+			blockEventPayload, _ := callEVMHeartBeat(t,
+				ctx,
+				vm,
+				snapshot)
+
+			require.NotEmpty(t, blockEventPayload.Hash)
+			require.Equal(t, uint64(21000), blockEventPayload.TotalGasUsed)
```

Comment:
can we also check the transaction hashes match the batch

Reply:
updated